cli/src/update/unix.rs

```diff
@@ -230,119 +184,57 @@ impl ApplicationUpdater {
-        let bash = self.download_file(bash_asset_url, shell_asset_name).await?;
-        let sig = self
-            .download_file(sig_asset_url, "phylum.update.minisig")
-            .await?;
-
-        debug!("Verifying the binary signature before move");
-        if !self.has_valid_signature(bin.as_str(), sig.as_str()) {
-            return Err(std::io::Error::new(
-                io::ErrorKind::Other,
-                "The update binary failed signature validation",
-            ));
-        }
+        let zip = download_github_asset(zip_asset).await?;
+        let sig = download_github_asset(sig_asset).await?;
 
-        // If the download and validation succeeds _then_ we move it to overwrite
-        // the existing binary and bash file.
-        debug!("Copying the files to the intended install location");
-        fs::remove_file(&installed_bin_path)?;
-        fs::copy(&bin, &installed_bin_path)?;
-        debug!(
-            "Copying shell script from {} to {:?}",
-            bash, installed_bash_path
-        );
-        fs::copy(&bash, &installed_bash_path)?;
-        debug!("Setting permissions ");
-        fs::set_permissions(&installed_bin_path, fs::Permissions::from_mode(0o770))?;
-        debug!("Removing bin");
-        fs::remove_file(&bin)?;
-        debug!("Removing shell script");
-        fs::remove_file(&bash)?;
-
-        // Ensure that the files copied to the final location were the ones
-        // we expected. This is to address a potential race condition between
-        // the check and the copy.
-        debug!("Verifying the file wasn't changed/tampered with before the move");
-        let final_bin = match installed_bin_path.clone().into_os_string().into_string() {
-            Ok(x) => x,
-            Err(_) => {
-                return Err(std::io::Error::new(
-                    io::ErrorKind::Other,
-                    "Could not create the path for the installation",
-                ))
-            }
-        };
+        debug!("Verifying the package signature");
+        if !self.has_valid_signature(zip.as_ref(), std::str::from_utf8(sig.as_ref())?) {
+            anyhow::bail!("The update binary failed signature validation");
+        }
 
-        if !self.has_valid_signature(final_bin.as_str(), sig.as_str()) {
-            fs::remove_file(&installed_bin_path)?;
-            fs::remove_file(&installed_bash_path)?;
+        debug!("Extracting package to temporary directory");
+        let temp_dir = tempfile::tempdir()?;
+        ZipArchive::new(std::io::Cursor::new(zip))?.extract(temp_dir.path())?;
 
-            return Err(std::io::Error::new(
-                io::ErrorKind::Other,
-                "Possible attack attempt! Binary changed after initial signature verification and was removed.",
-            ));
-        }
+        debug!("Running the installer");
+        let working_dir = temp_dir.path().join(archive_name);
+        let status = Command::new(working_dir.join("install.sh"))
```

Comment:
Going the safe route here makes sense, but just `"./install.sh"` should work since we're setting the working directory to `working_dir` anyway.

Reply:
That's true, but I like the safety of the full path for now.

After #216 is completed, it would probably be good to change this to `Command::new("/bin/sh").arg("install.sh")...` to avoid relying on the executable bit being set. As of now, this could fail if `/tmp` is mounted `noexec`